skrf/calibration/deembedding.py

```diff
@@ -1384,7 +2144,108 @@ def split2xthru(self, s2xthru):
 
         return (s_side1, s_side2)
 
-class IEEEP370_MM_NZC_2xThru(Deembedding):
+    def plot_check_residuals(self, ax: Axes = None) -> (Figure, Axes):
+        res = self.deembed(self.s2xthru)
+        res.name = 'Residuals'
+
+        if ax is None:
+            fig, ax = subplots(1, 2, sharex = True, figsize=(10, 5))
+        else:
+            fig = ax.get_figure()
+
+        fig.suptitle('Consistency test #1: Self de-embedding of 2X-Thru')
+
+        ax[0].set_title('Magnitude residuals')
+        res.plot_s_db(1,0, ax = ax[0], color = '0.5')
+        res.plot_s_db(0,1, ax = ax[0], color = 'k')
+        ax[0].plot([res.frequency.f_scaled[0], res.frequency.f_scaled[-1]],
+                       [0.1, 0.1],
+                       linestyle = 'dashed', color = 'r', label = 'Limit')
+        ax[0].plot([res.frequency.f_scaled[0], res.frequency.f_scaled[-1]],
+                       [-0.1, -0.1],
+                       linestyle = 'dashed', color = 'r')
+        ax[0].legend(loc = 'upper right')
+
+        ax[1].set_title('Phase residuals')
+        res.plot_s_deg(1,0, ax = ax[1], color = '0.5')
+        res.plot_s_deg(0,1, ax = ax[1], color = 'k')
+        ax[1].plot([res.frequency.f_scaled[0], res.frequency.f_scaled[-1]],
+                       [1, 1],
+                       linestyle = 'dashed', color = 'r', label = 'Limit')
+        ax[1].plot([res.frequency.f_scaled[0], res.frequency.f_scaled[-1]],
+                       [-1, -1],
+                       linestyle = 'dashed', color = 'r')
+        ax[1].legend(loc = 'upper right')
+        fig.tight_layout()
+
+        return (fig, ax)
+
+    def plot_check_impedance(self, fix_dut_fix: Network = None, ax: Axes = None,
+                             window: str = 'hamming') -> (Figure, Axes):
+        # if dc point already exists, it will be replaced
+        s2xthru = IEEEP370.extrapolate_to_dc(self.s2xthru)

```

Comment:
remark/question of Oriented Object programmers: as the IEEEP370 class should not be used (since it's an Abstract) shouldn't you use `self.` instead of `IEEEP370.` (and the method not defined as static) ?

Reply:
Thank you for this excellent question. Because the static function is inherited, I could maybe use `self` here 🤔 
Maybe I can give a short description of the programming challenge and gather your opinion about it:

- The user may want to extrapolate using IEEEP370 or test the fixtures electrical requirements without instantiating one of the deembedding classes this is why I used static initially.
- Currently, if one of the IEEEP370 classes is instantiated, the fixture is split in init which is time-consuming and requires suitable networks (so it cannot be instantiated to use the helpers functions only)
- Because of the split in init, the mixed modes flavours cannot call the init of the single-ended one, because the fixture split would be done twice.

I feel like if the split should have been performed in a `run(self)` method outside init but it would a major break the API.

Reply:
> Because the static function is inherited, I could maybe use self here 🤔

Well I cannot use `self` of course, but the class name, which is of little ehancement